AmazonChimeSDK/AmazonChimeSDK/audiovideo/video/backgroundfilter/BackgroundFilterProcessor.swift

```diff
@@ -0,0 +1,247 @@
+    /// has linked the necessary runtime framework (i.e. `AmazonChimeSDKMachineLearning`) to
+    /// use this class.
+    ///
+    /// - Returns: true if the class can be used, otherwise false.
+    public static func isAvailable() -> Bool {
+        return TensorFlowSegmentationProcessor.isAvailable()
+    }
+
+    /// Public constructor to initialize the processor.
+    ///
+    /// - Parameters:
+    ///   - logger: Custom logger to log events.
+    public init(logger: Logger) {
+        self.logger = logger
+        if !TensorFlowSegmentationProcessor.isAvailable() {
+            self.logger.error(msg:"Unable to load TensorFlowLiteSegmentationProcessor. " +
+                                "See `Update Project File` section in README for more information " +
+                                "on how to import `AmazonChimeSDKMachineLearning` framework " +
+                                "and the `selfie_segmentation_landscape.tflite` as a bundle resource " +
+                                "to your project.")
+            segmentationProcessor = NoopSegmentationProcessor()
+        } else {
+            segmentationProcessor = TensorFlowSegmentationProcessor()
+        }
+    }
+
+    /// Creates the alpha mask [0-255] of the foreground image using `SegmentationProcessor`.
+    ///
+    /// - Parameters:
+    ///   - inputFrameCG: Input CGImage frame to produce the foreground image.
+    ///   - inputFrameCI: Input CIImage frame to produce the foreground image.
+    ///
+    /// - Returns: Alpha mask CGImage of the foreground.
+    public func createForegroundAlphaMask(inputFrameCG: CGImage,
+                                          inputFrameCI: CIImage) -> CIImage? {
+        // Verify that the processor is available.
+        if !TensorFlowSegmentationProcessor.isAvailable() {
+            return nil
+        }
+
+        // Number of the input image color space channels.
+        let imageChannels = inputFrameCG.bitsPerPixel / inputFrameCG.bitsPerComponent
+
+        // Update the buffer pool dimensions if the new frame does not match the previous frame dimensions.
+        if bufferPool == nil || inputFrameCG.width != bufferPoolWidth || inputFrameCG.height != bufferPoolHeight {
+
+            updateBufferPool(newWidth: inputFrameCG.width, newHeight: inputFrameCG.height)
+            // Initialize the segmentationProcessor if it has not been initialized.
+            let initializeResult: Bool = segmentationProcessor.initialize(height: segmentationProcessorHeight,
+                                                     width: segmentationProcessorWidth,
+                                                     channels: imageChannels)
+            if !initializeResult {
+                logger.error(msg: "Unable to initialize segmentation processor.")
+                return nil
+            }
+        }
+
+        // Check if segmentation model has loaded.
+        if segmentationProcessor.getModelState() != 2 {
```

Comment:
Should create enum or constants to match ModelState instead of use raw value `2`.